### PR TITLE
command line for claimer

### PR DESCRIPTION
**Pull request by elv-Tom**

### CODE REVIEW

File: src/ElvContracts.js [hunk 1]
```diff
@@ -0,0 +1,215 @@
...
+    var lengthList = await this.client.CallContractMethod({
+      contractAddress: Config.consts.main.claimerAddress,
+      abi: JSON.parse(abi),
+      methodName: "getNrAllocations",
+      methodArgs: [ address ],
+      formatArguments: true,
+    });
+
+    let listAllocations = [];
+    for (let i ; i<lengthList; i++){
```
Comment (elv-wayne) [inline src/ElvContracts.js:182]: Add initializer in the for loop: let i=0;

File: src/ElvContracts.js [hunk 2]
```diff
@@ -0,0 +1,215 @@
...
+          formatArguments: true,
+        });
+        var elemExpirationDate = await this.client.CallContractMethod({
+          contractAddress: Config.consts.main.claimerAddress,
+          abi: JSON.parse(abi),
+          methodName: "getExpirationDate",
+          methodArgs: [ address, idx ],
+          formatArguments: true,
+        });
+        listAllocations.push("amount : " + elemAmount + " - expiration : " + elemExpirationDate);
```
Comment (elv-wayne) [inline src/ElvContracts.js:199]: Push a new object instead:  {amount: elementAmount, expiration: elemExpirationDate}

File: src/ElvContracts.js [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+      } catch (e){
+        break;
+      }
+    }
+    /**
+     * questions :
+     * -we return a string ?
+     */
+    return listAllocations.toString();
```
Comment (elv-wayne) [inline src/ElvContracts.js:208]: Just return the listAllocations

File: src/ElvContracts.js [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+     * @param {string} amount : amount to allocate
+     * @param {string} expirationDate : the expiration date of the new allocation
+     */
+  async ClaimerAllocate({ address, amount, expirationDate }){
+    const abi = fs.readFileSync(
+      path.resolve(__dirname, "../contracts/v4/Claimer.abi")
+    );
+
+    var res = await this.client.CallContractMethodAndWait({
+      contractAddress: Config.consts.main.claimerAddress,
```
Comment (elv-wayne) [inline src/ElvContracts.js:46]: Should use command params for the contract address and not the config

File: utilities/EluvioCli.js [hunk 5]
```diff
@@ -265,6 +266,185 @@ const CmdAccountSignedToken = async ({ argv }) => {
   }
 };
 
+const CmdClaimerAllocate = async ({ argv }) => {
+  console.log("Claimer Allocate\n");
+  console.log("args", argv);
+
+  try {
+    let elvAccount = new ElvContracts({
```
Comment (elv-wayne) [inline utilities/EluvioCli.js:274]: rename variable to elvContracts (For the other functions as well), confusing since we have an ElvAccount class

**Review by elv-wayne — CHANGES_REQUESTED** [review]

**Review by elv-serban — APPROVED** [review]
Looks good - ready to go

**Review by elv-wayne — APPROVED** [review]
I see the claimerAddress is still in the config instead of taking it as a command param but I'll leave it to you @elv-serban to decide

**Pull request merged**